### PR TITLE
[Backport stable/8.2] deps(go): bump go from 1.17 to 1.19

### PR DESCRIPTION
**Pull request by abbasadel**
Backport of https://github.com/camunda/zeebe/pull/12633 to stable/8.2.

relates to https://github.com/camunda/zeebe/issues/12649

### CODE REVIEW

**Comment by abbasadel** [comment]
bors merge

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -254,17 +254,14 @@ jobs:
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
       - uses: ./.github/actions/build-zeebe
         id: build-zeebe
-      # Once we're on Go 1.18, use the official gorelease to do this
+      # Once we have breaking changes, this will break since no workaround atm.
+      # Give the official gorelease a try to do this
```
Comment (Sijoma) [inline .github/workflows/ci.yml:258]: offical go release of the go-apidiff pkg I assume? I think joelanford is quite trusty. From the comments I cannot tell why the official one was not used. But I guess its fine! 

I'm happy that there is progress here! 💪

**Review by Sijoma — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Sijoma) [reply .github/workflows/ci.yml:258]: And now I know that I'm allowed to approve PR's here 🤯

Comment (abbasadel) [reply .github/workflows/ci.yml:258]: Thanks @Sijoma

**Comment by abbasadel** [comment]
bors merge

**Comment by npepinpe** [comment]
I've added you as a bors reviewer :+1:

**Comment by npepinpe** [comment]
bors merge

**Comment by abbasadel** [comment]
bros merge

**Comment by abbasadel** [comment]
bors merge

**Comment by abbasadel** [comment]
bors merge

**Pull request merged**